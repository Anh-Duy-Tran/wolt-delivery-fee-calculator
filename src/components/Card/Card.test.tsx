import { expect, it, describe } from 'vitest';
import { render, screen } from '@testing-library/react';
import { Card } from '.';

describe('Card Component', () => {
  // Snapshot test
  it('renders correctly', () => {
    const { asFragment } = render(<Card>Cart Title</Card>);
    expect(asFragment()).toMatchSnapshot();
  });

  // Test for class application
  it('applies default and custom classes', () => {
    render(<Card className="custom-class" />);
    const cardElement = screen.getByTestId('card');
    expect(cardElement).toHaveClass(
      'w-full rounded-xl bg-white p-5 shadow-xl custom-class'
    );
  });

  // Test for props spreading
  it('spreads additional props to the div element', () => {
    render(<Card aria-label="Test Card" />);
    const cardElement = screen.getByTestId('card');
    expect(cardElement).toHaveAttribute('aria-label', 'Test Card');
  });

  // Test for children rendering
  it('renders children correctly', () => {
    render(
      <Card>
        <p>Test Child</p>
      </Card>
    );
    expect(screen.getByText('Test Child')).toBeInTheDocument();
  });
});
