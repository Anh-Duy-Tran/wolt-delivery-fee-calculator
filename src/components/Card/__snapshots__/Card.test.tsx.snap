// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card Component > renders correctly 1`] = `
<DocumentFragment>
  <div
    class="w-full rounded-xl bg-white p-5 shadow-xl"
    data-testid="card"
  >
    Cart Title
  </div>
</DocumentFragment>
`;
